src/ProjectSystem/Microsoft.NET.Sdk.Web.ProjectSystem.Targets/netstandard1.0/Microsoft.NET.Sdk.Web.ProjectSystem.targets

```diff
@@ -20,36 +20,47 @@ Copyright (c) .NET Foundation. All rights reserved.
   <PropertyGroup>
     <EnableDefaultContentItems Condition=" '$(EnableDefaultContentItems)' == '' ">true</EnableDefaultContentItems>
     <RunWorkingDirectory Condition=" '$(RunWorkingDirectory)' == '' and '$(EnableDefaultRunWorkingDirectory)' != 'false' ">$(MSBuildProjectDirectory)</RunWorkingDirectory>
+    <MSBuildWebTargetsPath>$(MSBuildExtensionsPath)\Microsoft\VisualStudio\Managed.Web\</MSBuildWebTargetsPath>
   </PropertyGroup>
 
-  <PropertyGroup>
-    <MSBuildWebPath>$(MSBuildExtensionsPath)\Microsoft\VisualStudio\Managed.Web\</MSBuildWebPath>
-    <ActualLangName Condition="Exists('$(MSBuildWebPath)$(LangName)')">$(LangName)</ActualLangName>
-    <ActualLangName Condition="'$(ActualLangName)' == ''">en-us</ActualLangName>
-  </PropertyGroup>
+  <!--
+    Newer versions of Visual Studio ship the designtime related properties in a targets file and all future design time only elements should be added there. If that file does not
+    exist, if falls back to the default set of values defined here.
+  -->
+    <Import Project="$(MSBuildWebTargetsPath)Microsoft.Web.Designtime.targets" Condition="Exists('$(MSBuildWebTargetsPath)Microsoft.Web.Designtime.targets')" />
+
+    <Choose>
+      <When Condition="!Exists('$(MSBuildWebTargetsPath)Microsoft.Web.Designtime.targets')" >
+        <PropertyGroup>
+          <ActualLangName Condition="Exists('$(MSBuildWebTargetsPath)$(LangName)')">$(LangName)</ActualLangName>
+          <ActualLangName Condition="'$(ActualLangName)' == ''">en-us</ActualLangName>
+        </PropertyGroup>
+
+        <!-- Web project capabilities that enables web features for .NET Core projects -->
+        <ItemGroup >
+          <ProjectCapability Include="DotNetCoreWeb" />
```

Comment:
What happens if the version of Microsoft.Web.Designtime.targets is older than this targets file? i.e. someone installs a new .NET Core SDK. Wouldn't this mean new features of that SDK may not light up?

Can ProjectCapability be moved outside the Choose/When block? I believe VS will de-dup capabilities, so it should be okay if Microsoft.Web.Designtime.targets re-defines some of these.

Reply:
No new capabilities should be added to this file. Visual studio will ship any new capabilities that it supports (expect of course those delivered in other targets files, like from nuget).  I had to leave these ones here so that older versions of visual studio which don't have the design time targets file continue to work.

Reply:
Ok, thanks for clarifying.